joblib/test/test_hashing.py

```diff
@@ -70,8 +72,8 @@ def f(self, x):
 
 class KlassWithCachedMethod(object):
 
-    def __init__(self):
-        mem = Memory(cachedir=test_memory_env['dir'])
+    def __init__(self, tmpdir_path):
```

Comment:
Fair enough, although a little bit magical. We'll get used to it after some time.